src/composed/key/shared.rs

```diff
@@ -12,71 +12,47 @@ use crate::{
         UserAttribute, UserId,
     },
     ser::Serialize,
-    types::{
-        CompressionAlgorithm, KeyVersion, Password, PublicKeyTrait, RevocationKey, SecretKeyTrait,
-    },
+    types::{CompressionAlgorithm, KeyVersion, Password, PublicKeyTrait, SecretKeyTrait},
 };
 
+/// A KeyDetails specifies associated user id and attribute components, and some metadata for
```

Comment:
please write This or sth like that, repeating the type is a horrible convention golang made popular for some unknown confused reason